<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.0.36">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>RNA editing project - Data exploration patient 5</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="../../styles.css">
</head>

<body class="nav-sidebar docked nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">RNA editing project</span>
  </a>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../index.html">Home</a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../about.html">About</a>
  </li>  
</ul>
              <div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
    <div class="container-fluid d-flex justify-content-between">
      <h1 class="quarto-secondary-nav-title">Data exploration patient 5</h1>
      <button type="button" class="quarto-btn-toggle btn" aria-label="Show secondary navigation">
        <i class="bi bi-chevron-right"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse sidebar-navigation docked overflow-auto">
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">IBD dataset</a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../IBD/results/01A_Seurat_import_clean.html" class="sidebar-item-text sidebar-link">Seurat cleaning</a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">Crohn disease dataset</a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="false">COTAN cleaning</a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="false">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth2 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../Crohn_disease/results/01_cleaning_patient5_NB.html" class="sidebar-item-text sidebar-link">patient 5 ileum</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../Crohn_disease/results/01_cleaning_patient7_NB.html" class="sidebar-item-text sidebar-link">patient 7 ileum</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../Crohn_disease/results/01_cleaning_patient8_NB.html" class="sidebar-item-text sidebar-link">patient 8 ileum</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../Crohn_disease/results/01_cleaning_patient10_NB.html" class="sidebar-item-text sidebar-link">patient 10 ileum</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../Crohn_disease/results/01_cleaning_patient11_NB.html" class="sidebar-item-text sidebar-link">patient 11 ileum</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../Crohn_disease/results/01_cleaning_patient12_NB.html" class="sidebar-item-text sidebar-link">patient 12 ileum</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../Crohn_disease/results/01_cleaning_patient14_NB.html" class="sidebar-item-text sidebar-link">patient 14 ileum</a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="true">Data exploration &amp; Cluster identification</a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="true">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth2 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../Crohn_disease/results/04_Data_exploration_clusters_id_patient5_NB.html" class="sidebar-item-text sidebar-link active">patient 5 ileum</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../Crohn_disease/results/04_Data_exploration_clusters_id_patient7_NB.html" class="sidebar-item-text sidebar-link">patient 7 ileum</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../Crohn_disease/results/04_Data_exploration_clusters_id_patient8_NB.html" class="sidebar-item-text sidebar-link">patient 8 ileum</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../Crohn_disease/results/04_Data_exploration_clusters_id_patient10_NB.html" class="sidebar-item-text sidebar-link">patient 10 ileum</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../Crohn_disease/results/04_Data_exploration_clusters_id_patient11_NB.html" class="sidebar-item-text sidebar-link">patient 11 ileum</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../Crohn_disease/results/04_Data_exploration_clusters_id_patient12_NB.html" class="sidebar-item-text sidebar-link">patient 12 ileum</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../Crohn_disease/results/04_Data_exploration_clusters_id_patient14_NB.html" class="sidebar-item-text sidebar-link">patient 14 ileum</a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" aria-expanded="false">Summary and Conclusion</a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" aria-expanded="false">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth2 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../Crohn_disease/results/05_HeatmapSummary_NB.html" class="sidebar-item-text sidebar-link">Heatmaps summary</a>
  </div>
</li>
      </ul>
  </li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#import-markers" id="toc-import-markers" class="nav-link active" data-scroll-target="#import-markers">Import markers</a></li>
  <li><a href="#uninvolved-ileum-samples" id="toc-uninvolved-ileum-samples" class="nav-link" data-scroll-target="#uninvolved-ileum-samples">Uninvolved ileum samples</a>
  <ul class="collapse">
  <li><a href="#gdi-plot-with-isg-genes" id="toc-gdi-plot-with-isg-genes" class="nav-link" data-scroll-target="#gdi-plot-with-isg-genes">GDI plot with ISG genes</a></li>
  <li><a href="#cluster-assignment-with-expression-increment" id="toc-cluster-assignment-with-expression-increment" class="nav-link" data-scroll-target="#cluster-assignment-with-expression-increment">Cluster assignment with expression increment</a></li>
  <li><a href="#final-score-heatmap" id="toc-final-score-heatmap" class="nav-link" data-scroll-target="#final-score-heatmap">Final score heatmap</a></li>
  </ul></li>
  <li><a href="#involved-ileum-samples" id="toc-involved-ileum-samples" class="nav-link" data-scroll-target="#involved-ileum-samples">Involved ileum samples</a>
  <ul class="collapse">
  <li><a href="#gdi-plot-with-isg-genes-1" id="toc-gdi-plot-with-isg-genes-1" class="nav-link" data-scroll-target="#gdi-plot-with-isg-genes-1">GDI plot with ISG genes</a></li>
  <li><a href="#section" id="toc-section" class="nav-link" data-scroll-target="#section"></a></li>
  <li><a href="#final-score-heatmap-1" id="toc-final-score-heatmap-1" class="nav-link" data-scroll-target="#final-score-heatmap-1">Final score heatmap</a></li>
  </ul></li>
  <li><a href="#pbmc-samples" id="toc-pbmc-samples" class="nav-link" data-scroll-target="#pbmc-samples">PBMC samples</a>
  <ul class="collapse">
  <li><a href="#gdi-plot-with-isg-genes-2" id="toc-gdi-plot-with-isg-genes-2" class="nav-link" data-scroll-target="#gdi-plot-with-isg-genes-2">GDI plot with ISG genes</a></li>
  <li><a href="#cluster-assignment-with-expression-increment-1" id="toc-cluster-assignment-with-expression-increment-1" class="nav-link" data-scroll-target="#cluster-assignment-with-expression-increment-1">Cluster assignment with expression increment</a></li>
  <li><a href="#final-score-heatmap-2" id="toc-final-score-heatmap-2" class="nav-link" data-scroll-target="#final-score-heatmap-2">Final score heatmap</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title d-none d-lg-block">Data exploration patient 5</h1>
</div>



<div class="quarto-title-meta">

    
    
  </div>
  

</header>

<div class="cell">
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: viridisLite</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
---------------------
Welcome to dendextend version 1.16.0
Type citation('dendextend') for how to cite the package.

Type browseVignettes(package = 'dendextend') for the package vignette.
The github page is: https://github.com/talgalili/dendextend/

Suggestions and bug-reports can be submitted at: https://github.com/talgalili/dendextend/issues
You may ask questions at stackoverflow, use the r and dendextend tags: 
     https://stackoverflow.com/questions/tagged/dendextend

    To suppress this message use:  suppressPackageStartupMessages(library(dendextend))
---------------------</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'dendextend'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:stats':

    cutree</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>ℹ Loading COTAN</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>today <span class="ot">&lt;-</span> <span class="fu">Sys.Date</span>()</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="fu">format</span>(today, <span class="at">format =</span> <span class="st">"%A %d %B %Y"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Thursday 22 December 2022"</code></pre>
</div>
</div>
<section id="import-markers" class="level3">
<h3 class="anchored" data-anchor-id="import-markers">Import markers</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>markers <span class="ot">&lt;-</span> readxl<span class="sc">::</span><span class="fu">read_excel</span>(<span class="st">"../../Marker_genes.xlsx"</span>,<span class="at">sheet =</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>New names:
• `` -&gt; `...1`</code></pre>
</div>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>markers <span class="ot">&lt;-</span> <span class="fu">as.list</span>(markers)</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>markers <span class="ot">&lt;-</span> markers[<span class="dv">2</span><span class="sc">:</span><span class="fu">length</span>(markers)]</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (nv <span class="cf">in</span> <span class="fu">names</span>(markers)) {</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>  markers[[nv]] <span class="ot">&lt;-</span> <span class="fu">unique</span>(markers[[nv]][<span class="sc">!</span><span class="fu">is.na</span>(markers[[nv]])])</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="uninvolved-ileum-samples" class="level1">
<h1>Uninvolved ileum samples</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>out_dir <span class="ot">&lt;-</span> <span class="st">"../../Crohn_disease/data/04_cluster_identification/"</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>in_dir <span class="ot">&lt;-</span> <span class="st">"../../Crohn_disease/data/02_cleaned_data/"</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>sample.id <span class="ot">=</span> <span class="st">"patient5_Uninvolved"</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>expression.cl <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">paste0</span>(out_dir,sample.id,<span class="st">"_expression_clusters.csv"</span>),<span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">paste0</span>(in_dir,sample.id,<span class="st">".clustered.merged.cotan.RDS"</span>))</span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>ISG.genes <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../../ISG_fixed.txt"</span>,<span class="at">header =</span> T)</span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(ISG.genes) <span class="ot">&lt;-</span> <span class="st">"genes"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Number of ISG genes</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(ISG.genes)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 332   1</code></pre>
</div>
</div>
<p>Detected ISG genes in the sample</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(ISG.genes[ISG.genes<span class="sc">$</span>genes <span class="sc">%in%</span> <span class="fu">getGenes</span>(obj), ,<span class="at">drop =</span> <span class="cn">FALSE</span>])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 203   1</code></pre>
</div>
</div>
<section id="gdi-plot-with-isg-genes" class="level3">
<h3 class="anchored" data-anchor-id="gdi-plot-with-isg-genes">GDI plot with ISG genes</h3>
<p>Gene Differentiation Index is a score of how much a gene is differentially expressed in the dataset. Genes expressed only in a subpopulation of cells will have a GDI score higher than 1.5 while constitutive genes will show a GDI lower than 1.5. In the plot on the x-axis, there is also an estimation of the mean gene expression in the whole sample.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>GDI.data <span class="ot">&lt;-</span> <span class="fu">calculateGDI</span>(<span class="at">objCOTAN =</span> obj)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Setting new log level to 1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">GDIPlot</span>(<span class="at">objCOTAN =</span> obj, </span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>        <span class="at">genes =</span> <span class="fu">list</span>(<span class="st">"High exp ISG"</span><span class="ot">=</span><span class="fu">rownames</span>(GDI.data[<span class="fu">rownames</span>(GDI.data) <span class="sc">%in%</span> </span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>                                  ISG.genes<span class="sc">$</span>genes <span class="sc">&amp;</span></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>                                  GDI.data<span class="sc">$</span>sum.raw.norm <span class="sc">&gt;=</span> <span class="dv">9</span>,]),</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>                          <span class="st">"High GDI ISG"</span><span class="ot">=</span><span class="fu">rownames</span>(GDI.data[<span class="fu">rownames</span>(GDI.data) <span class="sc">%in%</span></span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>                                                             ISG.genes<span class="sc">$</span>genes <span class="sc">&amp;</span></span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>                                  GDI.data<span class="sc">$</span>GDI <span class="sc">&gt;=</span> <span class="fu">quantile</span>(GDI.data<span class="sc">$</span>GDI,<span class="at">probs =</span> <span class="fl">0.99</span>),])),</span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">GDI.df =</span> GDI.data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="04_Data_exploration_clusters_id_patient5_NB_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid figure-img" width="672"></p>
<p></p><figcaption class="figure-caption">In green the most expressed ISG genes and in purple the top GDI ISG genes in the sample. Red dotted line indicates the 1.5 GDI threshold for differentially expressed genes. The two blue lines indicate the firs and senonc quartile for GDI score.</figcaption><p></p>
</figure>
</div>
</div>
</div>
</section>
<section id="cluster-assignment-with-expression-increment" class="level3">
<h3 class="anchored" data-anchor-id="cluster-assignment-with-expression-increment">Cluster assignment with expression increment</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>score.df <span class="ot">&lt;-</span> <span class="fu">geneSetEnrichment</span>(<span class="at">objCOTAN =</span> obj,</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>                              <span class="at">genes =</span> markers, </span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">expression.cl =</span> expression.cl)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "In Epithelial genes there are 21 detected over 42 genes"
[1] "In Enterocytes there are 2 detected over 4 genes"
[1] "In Goblet cells there are 1 detected over 6 genes"
[1] "In Enteroendocrine cells there are 0 detected over 4 genes"
[1] "In Paneth cells there are 1 detected over 3 genes"
[1] "In Tuft cells there are 0 detected over 3 genes"
[1] "In TA cells there are 2 detected over 5 genes"
[1] "In Stem cells there are 1 detected over 5 genes"
[1] "In BEST4 cells there are 1 detected over 4 genes"
[1] "In Red blood genes there are 3 detected over 3 genes"
[1] "In T cell genes there are 6 detected over 6 genes"
[1] "In B cell genes there are 6 detected over 6 genes"
[1] "In Plasma Cells there are 3 detected over 4 genes"
[1] "In Dendridic Cells there are 6 detected over 8 genes"
[1] "In Macrophages there are 4 detected over 5 genes"
[1] "In ILCs - Cytotoxic natural killer there are 6 detected over 8 genes"
[1] "In ILC1s there are 10 detected over 11 genes"
[1] "In ILC2s there are 6 detected over 11 genes"
[1] "In ILC3 there are 4 detected over 6 genes"
[1] "In Mast cells there are 4 detected over 6 genes"
[1] "In Endotelial Cells there are 4 detected over 7 genes"
[1] "In ?? there are 0 detected over 2 genes"
[1] "In Fibroblasts there are 3 detected over 6 genes"
[1] "In Smooth muscle cells there are 4 detected over 5 genes"
[1] "In Enteric glia cells there are 3 detected over 4 genes"
[1] "In Neutrophils there are 2 detected over 8 genes"
[1] "In Eosinophil there are 0 detected over 2 genes"
[1] "In Basophils there are 4 detected over 6 genes"</code></pre>
</div>
</div>
<p>We add separately the ISG score (to avoid the division of the score isubpopulationesent in more marker lists)</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>score.ISG <span class="ot">&lt;-</span> <span class="fu">geneSetEnrichment</span>(<span class="at">objCOTAN =</span> obj,</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>                              <span class="at">genes =</span> <span class="fu">list</span>(<span class="st">"ISG"</span><span class="ot">=</span>ISG.genes<span class="sc">$</span>genes), </span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">expression.cl =</span> expression.cl)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "In ISG there are 203 detected over 332 genes"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>score.df <span class="ot">&lt;-</span> <span class="fu">rbind</span>(score.df,score.ISG)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>In the next table we can observe how many marker genes we defined and how many are detected in thex-axis,e.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>score.df[,<span class="fu">c</span>(<span class="st">"N. detected"</span>,<span class="st">"N. total"</span>)]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                N. detected N. total
Epithelial genes                         21       42
Enterocytes                               2        4
Goblet cells                              1        6
Enteroendocrine cells                     0        4
Paneth cells                              1        3
Tuft cells                                0        3
TA cells                                  2        5
Stem cells                                1        5
BEST4 cells                               1        4
Red blood genes                           3        3
T cell genes                              6        6
B cell genes                              6        6
Plasma Cells                              3        4
Dendridic Cells                           6        8
Macrophages                               4        5
ILCs - Cytotoxic natural killer           6        8
ILC1s                                    10       11
ILC2s                                     6       11
ILC3                                      4        6
Mast cells                                4        6
Endotelial Cells                          4        7
??                                        0        2
Fibroblasts                               3        6
Smooth muscle cells                       4        5
Enteric glia cells                        3        4
Neutrophils                               2        8
Eosinophil                                0        2
Basophils                                 4        6
ISG                                     203      332</code></pre>
</div>
</div>
<p>Cluster tree using the cosine distance over cluster coex.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>cluster_data <span class="ot">&lt;-</span> <span class="fu">getClusterizationData</span>(obj)[[<span class="st">"coex"</span>]]</span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a><span class="do">######## This is the best: cosine dissimilarity</span></span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>Matrix <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">t</span>(cluster_data))</span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>sim <span class="ot">&lt;-</span> Matrix <span class="sc">/</span> <span class="fu">sqrt</span>(<span class="fu">rowSums</span>(Matrix <span class="sc">*</span> Matrix))</span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>sim <span class="ot">&lt;-</span> sim <span class="sc">%*%</span> <span class="fu">t</span>(sim)</span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a>D_sim <span class="ot">&lt;-</span> <span class="fu">as.dist</span>(<span class="dv">1</span> <span class="sc">-</span> sim)</span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a>tree <span class="ot">&lt;-</span> <span class="fu">hclust</span>(D_sim,<span class="at">method =</span> <span class="st">"ward.D2"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get_col(col, k): Length of color vector was shorter than the number
of clusters - color vector was recycled</code></pre>
</div>
</div>
<div class="cell">

</div>
<div class="cell">

</div>
</section>
<section id="final-score-heatmap" class="level3">
<h3 class="anchored" data-anchor-id="final-score-heatmap">Final score heatmap</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>col_fun <span class="ot">=</span> <span class="fu">colorRamp2</span>(<span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>), <span class="fu">c</span>( <span class="st">"lightblue"</span>, <span class="st">"red"</span>))</span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>final.heatmap <span class="ot">&lt;-</span> <span class="fu">Heatmap</span>(df.t, <span class="at">rect_gp =</span> <span class="fu">gpar</span>(<span class="at">col =</span> <span class="st">"white"</span>, <span class="at">lwd =</span> <span class="dv">1</span>),</span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>        <span class="co">#cluster_rows = F,</span></span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">cluster_columns =</span> <span class="cn">FALSE</span>,</span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">show_heatmap_legend =</span> F,</span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">col =</span> col_fun,</span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">width =</span>  <span class="fu">unit</span>(<span class="dv">28</span>, <span class="st">"cm"</span>),</span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">row_dend_width =</span> <span class="fu">unit</span>(<span class="dv">8</span>, <span class="st">"cm"</span>),</span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a>        <span class="co">#height = unit(6, "cm"),</span></span>
<span id="cb28-10"><a href="#cb28-10" aria-hidden="true" tabindex="-1"></a>        <span class="at">cluster_rows =</span> dend,</span>
<span id="cb28-11"><a href="#cb28-11" aria-hidden="true" tabindex="-1"></a>        <span class="at">column_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">11</span>),</span>
<span id="cb28-12"><a href="#cb28-12" aria-hidden="true" tabindex="-1"></a>      <span class="at">row_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">9</span>),</span>
<span id="cb28-13"><a href="#cb28-13" aria-hidden="true" tabindex="-1"></a>        <span class="at">cell_fun =</span> <span class="cf">function</span>(j, i, x, y, width, height, fill) {</span>
<span id="cb28-14"><a href="#cb28-14" aria-hidden="true" tabindex="-1"></a>        <span class="fu">grid.text</span>(<span class="fu">sprintf</span>(<span class="st">"%.1f"</span>, df.t[i, j]), x, y, <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">9</span>))</span>
<span id="cb28-15"><a href="#cb28-15" aria-hidden="true" tabindex="-1"></a>},</span>
<span id="cb28-16"><a href="#cb28-16" aria-hidden="true" tabindex="-1"></a><span class="at">right_annotation =</span> ha,</span>
<span id="cb28-17"><a href="#cb28-17" aria-hidden="true" tabindex="-1"></a><span class="at">left_annotation =</span> hb</span>
<span id="cb28-18"><a href="#cb28-18" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb28-19"><a href="#cb28-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-20"><a href="#cb28-20" aria-hidden="true" tabindex="-1"></a>final.heatmap</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="04_Data_exploration_clusters_id_patient5_NB_files/figure-html/unnamed-chunk-15-1.png" class="img-fluid" width="1536"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(final.heatmap,<span class="fu">paste0</span>(<span class="st">"../../Crohn_disease/data/04_cluster_identification/"</span>,sample.id,<span class="st">"_heatmap.RDS"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="involved-ileum-samples" class="level1">
<h1>Involved ileum samples</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>sample.id <span class="ot">=</span> <span class="st">"patient5_Involved"</span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a>expression.cl <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">paste0</span>(out_dir,sample.id,<span class="st">"_expression_clusters.csv"</span>),<span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">paste0</span>(in_dir,sample.id,<span class="st">".clustered.merged.cotan.RDS"</span>))</span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a>ISG.genes <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../../ISG_fixed.txt"</span>,<span class="at">header =</span> T)</span>
<span id="cb30-8"><a href="#cb30-8" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(ISG.genes) <span class="ot">&lt;-</span> <span class="st">"genes"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Number of ISG genes</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(ISG.genes)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 332   1</code></pre>
</div>
</div>
<p>Detected ISG genes in the sample</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(ISG.genes[ISG.genes<span class="sc">$</span>genes <span class="sc">%in%</span> <span class="fu">getGenes</span>(obj), ,<span class="at">drop =</span> <span class="cn">FALSE</span>])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 277   1</code></pre>
</div>
</div>
<section id="gdi-plot-with-isg-genes-1" class="level3">
<h3 class="anchored" data-anchor-id="gdi-plot-with-isg-genes-1">GDI plot with ISG genes</h3>
<p>Gene Differentiation Index is a score of how mutch a gene is different,tially expressed in the dataset. Genes expressed only in a sub-population of cells will have a GDI score higher than 1.5 while constitutive genes will show a GDI lower than 1.5. In the plot on the x axis there is also an estimation of the mean gene expression in the whole sample.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>GDI.data <span class="ot">&lt;-</span> <span class="fu">calculateGDI</span>(<span class="at">objCOTAN =</span> obj)</span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a><span class="fu">GDIPlot</span>(<span class="at">objCOTAN =</span> obj, </span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">genes =</span> <span class="fu">list</span>(<span class="st">"High exp ISG"</span><span class="ot">=</span><span class="fu">rownames</span>(GDI.data[<span class="fu">rownames</span>(GDI.data) <span class="sc">%in%</span> </span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a>                                  ISG.genes<span class="sc">$</span>genes <span class="sc">&amp;</span></span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a>                                  GDI.data<span class="sc">$</span>sum.raw.norm <span class="sc">&gt;=</span> <span class="dv">9</span>,]),</span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a>                          <span class="st">"High GDI ISG"</span><span class="ot">=</span><span class="fu">rownames</span>(GDI.data[<span class="fu">rownames</span>(GDI.data) <span class="sc">%in%</span></span>
<span id="cb35-7"><a href="#cb35-7" aria-hidden="true" tabindex="-1"></a>                                                             ISG.genes<span class="sc">$</span>genes <span class="sc">&amp;</span></span>
<span id="cb35-8"><a href="#cb35-8" aria-hidden="true" tabindex="-1"></a>                                  GDI.data<span class="sc">$</span>GDI <span class="sc">&gt;=</span> <span class="fu">quantile</span>(GDI.data<span class="sc">$</span>GDI,<span class="at">probs =</span> <span class="fl">0.99</span>),])),</span>
<span id="cb35-9"><a href="#cb35-9" aria-hidden="true" tabindex="-1"></a>        <span class="at">GDI.df =</span> GDI.data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="04_Data_exploration_clusters_id_patient5_NB_files/figure-html/unnamed-chunk-20-1.png" class="img-fluid figure-img" width="672"></p>
<p></p><figcaption class="figure-caption">In green the most expressed ISG genes and in purple the top GDI ISG genes in the sample. Red dotted line indicates the 1.5 GDI threshold for differentially expressed genes. The two blue lines indicate the firs and senonc quartile for GDI score.</figcaption><p></p>
</figure>
</div>
</div>
</div>
</section>
<section id="section" class="level3">
<h3 class="anchored" data-anchor-id="section"></h3>
<p>Cluster assignment with expression increment</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>score.df <span class="ot">&lt;-</span> <span class="fu">geneSetEnrichment</span>(<span class="at">objCOTAN =</span> obj,</span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>                              <span class="at">genes =</span> markers, </span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">expression.cl =</span> expression.cl)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "In Epithelial genes there are 6 detected over 42 genes"
[1] "In Enterocytes there are 0 detected over 4 genes"
[1] "In Goblet cells there are 0 detected over 6 genes"
[1] "In Enteroendocrine cells there are 1 detected over 4 genes"
[1] "In Paneth cells there are 1 detected over 3 genes"
[1] "In Tuft cells there are 0 detected over 3 genes"
[1] "In TA cells there are 5 detected over 5 genes"
[1] "In Stem cells there are 3 detected over 5 genes"
[1] "In BEST4 cells there are 1 detected over 4 genes"
[1] "In Red blood genes there are 3 detected over 3 genes"
[1] "In T cell genes there are 6 detected over 6 genes"
[1] "In B cell genes there are 6 detected over 6 genes"
[1] "In Plasma Cells there are 3 detected over 4 genes"
[1] "In Dendridic Cells there are 7 detected over 8 genes"
[1] "In Macrophages there are 5 detected over 5 genes"
[1] "In ILCs - Cytotoxic natural killer there are 7 detected over 8 genes"
[1] "In ILC1s there are 10 detected over 11 genes"
[1] "In ILC2s there are 7 detected over 11 genes"
[1] "In ILC3 there are 5 detected over 6 genes"
[1] "In Mast cells there are 3 detected over 6 genes"
[1] "In Endotelial Cells there are 6 detected over 7 genes"
[1] "In ?? there are 1 detected over 2 genes"
[1] "In Fibroblasts there are 5 detected over 6 genes"
[1] "In Smooth muscle cells there are 5 detected over 5 genes"
[1] "In Enteric glia cells there are 2 detected over 4 genes"
[1] "In Neutrophils there are 5 detected over 8 genes"
[1] "In Eosinophil there are 0 detected over 2 genes"
[1] "In Basophils there are 3 detected over 6 genes"</code></pre>
</div>
</div>
<p>We add separately the ISG score (to avoid the division of the score if a gene is present in more marker lists)</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a>score.ISG <span class="ot">&lt;-</span> <span class="fu">geneSetEnrichment</span>(<span class="at">objCOTAN =</span> obj,</span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a>                              <span class="at">genes =</span> <span class="fu">list</span>(<span class="st">"ISG"</span><span class="ot">=</span>ISG.genes<span class="sc">$</span>genes), </span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">expression.cl =</span> expression.cl)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "In ISG there are 277 detected over 332 genes"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a>score.df <span class="ot">&lt;-</span> <span class="fu">rbind</span>(score.df,score.ISG)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>In the next table we can observe how many marker genes we defined and how many are detecx-axis the sample.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a>score.df[,<span class="fu">c</span>(<span class="st">"N. detected"</span>,<span class="st">"N. total"</span>)]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                N. detected N. total
Epithelial genes                          6       42
Enterocytes                               0        4
Goblet cells                              0        6
Enteroendocrine cells                     1        4
Paneth cells                              1        3
Tuft cells                                0        3
TA cells                                  5        5
Stem cells                                3        5
BEST4 cells                               1        4
Red blood genes                           3        3
T cell genes                              6        6
B cell genes                              6        6
Plasma Cells                              3        4
Dendridic Cells                           7        8
Macrophages                               5        5
ILCs - Cytotoxic natural killer           7        8
ILC1s                                    10       11
ILC2s                                     7       11
ILC3                                      5        6
Mast cells                                3        6
Endotelial Cells                          6        7
??                                        1        2
Fibroblasts                               5        6
Smooth muscle cells                       5        5
Enteric glia cells                        2        4
Neutrophils                               5        8
Eosinophil                                0        2
Basophils                                 3        6
ISG                                     277      332</code></pre>
</div>
</div>
<p>Cluster tree using the cosine distance over cluster coex.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a>cluster_data <span class="ot">&lt;-</span> <span class="fu">getClusterizationData</span>(obj)[[<span class="st">"coex"</span>]]</span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a><span class="do">######## This is the best: cosine dissimilarity</span></span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a>Matrix <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">t</span>(cluster_data))</span>
<span id="cb43-5"><a href="#cb43-5" aria-hidden="true" tabindex="-1"></a>sim <span class="ot">&lt;-</span> Matrix <span class="sc">/</span> <span class="fu">sqrt</span>(<span class="fu">rowSums</span>(Matrix <span class="sc">*</span> Matrix))</span>
<span id="cb43-6"><a href="#cb43-6" aria-hidden="true" tabindex="-1"></a>sim <span class="ot">&lt;-</span> sim <span class="sc">%*%</span> <span class="fu">t</span>(sim)</span>
<span id="cb43-7"><a href="#cb43-7" aria-hidden="true" tabindex="-1"></a>D_sim <span class="ot">&lt;-</span> <span class="fu">as.dist</span>(<span class="dv">1</span> <span class="sc">-</span> sim)</span>
<span id="cb43-8"><a href="#cb43-8" aria-hidden="true" tabindex="-1"></a>tree <span class="ot">&lt;-</span> <span class="fu">hclust</span>(D_sim,<span class="at">method =</span> <span class="st">"ward.D2"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get_col(col, k): Length of color vector was shorter than the number
of clusters - color vector was recycled</code></pre>
</div>
</div>
<div class="cell">

</div>
<div class="cell">

</div>
</section>
<section id="final-score-heatmap-1" class="level3">
<h3 class="anchored" data-anchor-id="final-score-heatmap-1">Final score heatmap</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a>col_fun <span class="ot">=</span> <span class="fu">colorRamp2</span>(<span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>), <span class="fu">c</span>( <span class="st">"lightblue"</span>, <span class="st">"red"</span>))</span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a>final.heatmap <span class="ot">&lt;-</span> <span class="fu">Heatmap</span>(df.t, <span class="at">rect_gp =</span> <span class="fu">gpar</span>(<span class="at">col =</span> <span class="st">"white"</span>, <span class="at">lwd =</span> <span class="dv">1</span>),</span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a>        <span class="co">#cluster_rows = F,</span></span>
<span id="cb45-4"><a href="#cb45-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">cluster_columns =</span> <span class="cn">FALSE</span>,</span>
<span id="cb45-5"><a href="#cb45-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">show_heatmap_legend =</span> F,</span>
<span id="cb45-6"><a href="#cb45-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">col =</span> col_fun,</span>
<span id="cb45-7"><a href="#cb45-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">width =</span>  <span class="fu">unit</span>(<span class="dv">28</span>, <span class="st">"cm"</span>),</span>
<span id="cb45-8"><a href="#cb45-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">row_dend_width =</span> <span class="fu">unit</span>(<span class="dv">8</span>, <span class="st">"cm"</span>),</span>
<span id="cb45-9"><a href="#cb45-9" aria-hidden="true" tabindex="-1"></a>        <span class="co">#height = unit(6, "cm"),</span></span>
<span id="cb45-10"><a href="#cb45-10" aria-hidden="true" tabindex="-1"></a>        <span class="at">cluster_rows =</span> dend,</span>
<span id="cb45-11"><a href="#cb45-11" aria-hidden="true" tabindex="-1"></a>        <span class="at">column_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">11</span>),</span>
<span id="cb45-12"><a href="#cb45-12" aria-hidden="true" tabindex="-1"></a>      <span class="at">row_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">9</span>),</span>
<span id="cb45-13"><a href="#cb45-13" aria-hidden="true" tabindex="-1"></a>        <span class="at">cell_fun =</span> <span class="cf">function</span>(j, i, x, y, width, height, fill) {</span>
<span id="cb45-14"><a href="#cb45-14" aria-hidden="true" tabindex="-1"></a>        <span class="fu">grid.text</span>(<span class="fu">sprintf</span>(<span class="st">"%.1f"</span>, df.t[i, j]), x, y, <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">9</span>))</span>
<span id="cb45-15"><a href="#cb45-15" aria-hidden="true" tabindex="-1"></a>},</span>
<span id="cb45-16"><a href="#cb45-16" aria-hidden="true" tabindex="-1"></a><span class="at">right_annotation =</span> ha,</span>
<span id="cb45-17"><a href="#cb45-17" aria-hidden="true" tabindex="-1"></a><span class="at">left_annotation =</span> hb</span>
<span id="cb45-18"><a href="#cb45-18" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb45-19"><a href="#cb45-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-20"><a href="#cb45-20" aria-hidden="true" tabindex="-1"></a>final.heatmap</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="04_Data_exploration_clusters_id_patient5_NB_files/figure-html/unnamed-chunk-28-1.png" class="img-fluid" width="1536"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(final.heatmap,<span class="fu">paste0</span>(<span class="st">"../../Crohn_disease/data/04_cluster_identification/"</span>,sample.id,<span class="st">"_heatmap.RDS"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="pbmc-samples" class="level1">
<h1>PBMC samples</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a>sample.id <span class="ot">=</span> <span class="st">"patient5_PBMC"</span></span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a>expression.cl <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">paste0</span>(out_dir,sample.id,<span class="st">"_expression_clusters.csv"</span>),<span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">paste0</span>(in_dir,sample.id,<span class="st">".clustered.merged.cotan.RDS"</span>))</span>
<span id="cb47-5"><a href="#cb47-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-6"><a href="#cb47-6" aria-hidden="true" tabindex="-1"></a>ISG.genes <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../../ISG_fixed.txt"</span>,<span class="at">header =</span> T)</span>
<span id="cb47-7"><a href="#cb47-7" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(ISG.genes) <span class="ot">&lt;-</span> <span class="st">"genes"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Number of ISG genes</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(ISG.genes)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 332   1</code></pre>
</div>
</div>
<p>Detected ISG genes in the sample</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(ISG.genes[ISG.genes<span class="sc">$</span>genes <span class="sc">%in%</span> <span class="fu">getGenes</span>(obj), ,<span class="at">drop =</span> <span class="cn">FALSE</span>])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 184   1</code></pre>
</div>
</div>
<section id="gdi-plot-with-isg-genes-2" class="level3">
<h3 class="anchored" data-anchor-id="gdi-plot-with-isg-genes-2">GDI plot with ISG genes</h3>
<p>Gene Differentiation Index is a score of how mutch a gene is differentially expressed in the dataset. Genes expressed only in a subpopulation of cells will have a GDI score higher than 1.5 while constitutive genes will show a GDI lower than 1.5. In the plot on the x axis there is also an estimation of the mean gene expression in the whole sample.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a>GDI.data <span class="ot">&lt;-</span> <span class="fu">calculateGDI</span>(<span class="at">objCOTAN =</span> obj)</span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a><span class="fu">GDIPlot</span>(<span class="at">objCOTAN =</span> obj, </span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">genes =</span> <span class="fu">list</span>(<span class="st">"High exp ISG"</span><span class="ot">=</span><span class="fu">rownames</span>(GDI.data[<span class="fu">rownames</span>(GDI.data) <span class="sc">%in%</span> </span>
<span id="cb52-4"><a href="#cb52-4" aria-hidden="true" tabindex="-1"></a>                                  ISG.genes<span class="sc">$</span>genes <span class="sc">&amp;</span></span>
<span id="cb52-5"><a href="#cb52-5" aria-hidden="true" tabindex="-1"></a>                                  GDI.data<span class="sc">$</span>sum.raw.norm <span class="sc">&gt;=</span> <span class="dv">9</span>,]),</span>
<span id="cb52-6"><a href="#cb52-6" aria-hidden="true" tabindex="-1"></a>                          <span class="st">"High GDI ISG"</span><span class="ot">=</span><span class="fu">rownames</span>(GDI.data[<span class="fu">rownames</span>(GDI.data) <span class="sc">%in%</span></span>
<span id="cb52-7"><a href="#cb52-7" aria-hidden="true" tabindex="-1"></a>                                                             ISG.genes<span class="sc">$</span>genes <span class="sc">&amp;</span></span>
<span id="cb52-8"><a href="#cb52-8" aria-hidden="true" tabindex="-1"></a>                                  GDI.data<span class="sc">$</span>GDI <span class="sc">&gt;=</span> <span class="fu">quantile</span>(GDI.data<span class="sc">$</span>GDI,<span class="at">probs =</span> <span class="fl">0.99</span>),])),</span>
<span id="cb52-9"><a href="#cb52-9" aria-hidden="true" tabindex="-1"></a>        <span class="at">GDI.df =</span> GDI.data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="04_Data_exploration_clusters_id_patient5_NB_files/figure-html/unnamed-chunk-33-1.png" class="img-fluid figure-img" width="672"></p>
<p></p><figcaption class="figure-caption">In green the most expressed ISG genes and in purple the top GDI ISG genes in the sample. Red dotted line indicates the 1.5 GDI threshold for differentially expressed genes. The two blue lines indicate the firs and senonc quartile for GDI score.</figcaption><p></p>
</figure>
</div>
</div>
</div>
</section>
<section id="cluster-assignment-with-expression-increment-1" class="level3">
<h3 class="anchored" data-anchor-id="cluster-assignment-with-expression-increment-1">Cluster assignment with expression increment</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a>score.df <span class="ot">&lt;-</span> <span class="fu">geneSetEnrichment</span>(<span class="at">objCOTAN =</span> obj,</span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a>                              <span class="at">genes =</span> markers, </span>
<span id="cb53-3"><a href="#cb53-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">expression.cl =</span> expression.cl)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "In Epithelial genes there are 2 detected over 42 genes"
[1] "In Enterocytes there are 0 detected over 4 genes"
[1] "In Goblet cells there are 0 detected over 6 genes"
[1] "In Enteroendocrine cells there are 0 detected over 4 genes"
[1] "In Paneth cells there are 1 detected over 3 genes"
[1] "In Tuft cells there are 0 detected over 3 genes"
[1] "In TA cells there are 2 detected over 5 genes"
[1] "In Stem cells there are 1 detected over 5 genes"
[1] "In BEST4 cells there are 1 detected over 4 genes"
[1] "In Red blood genes there are 3 detected over 3 genes"
[1] "In T cell genes there are 6 detected over 6 genes"
[1] "In B cell genes there are 6 detected over 6 genes"
[1] "In Plasma Cells there are 2 detected over 4 genes"
[1] "In Dendridic Cells there are 7 detected over 8 genes"
[1] "In Macrophages there are 5 detected over 5 genes"
[1] "In ILCs - Cytotoxic natural killer there are 8 detected over 8 genes"
[1] "In ILC1s there are 10 detected over 11 genes"
[1] "In ILC2s there are 5 detected over 11 genes"
[1] "In ILC3 there are 1 detected over 6 genes"
[1] "In Mast cells there are 1 detected over 6 genes"
[1] "In Endotelial Cells there are 2 detected over 7 genes"
[1] "In ?? there are 0 detected over 2 genes"
[1] "In Fibroblasts there are 1 detected over 6 genes"
[1] "In Smooth muscle cells there are 3 detected over 5 genes"
[1] "In Enteric glia cells there are 2 detected over 4 genes"
[1] "In Neutrophils there are 5 detected over 8 genes"
[1] "In Eosinophil there are 0 detected over 2 genes"
[1] "In Basophils there are 2 detected over 6 genes"</code></pre>
</div>
</div>
<p>We add separately the ISG score (to avoid the division of the score is a gene is present in more marker list)</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a>score.ISG <span class="ot">&lt;-</span> <span class="fu">geneSetEnrichment</span>(<span class="at">objCOTAN =</span> obj,</span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a>                              <span class="at">genes =</span> <span class="fu">list</span>(<span class="st">"ISG"</span><span class="ot">=</span>ISG.genes<span class="sc">$</span>genes), </span>
<span id="cb55-3"><a href="#cb55-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">expression.cl =</span> expression.cl)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "In ISG there are 184 detected over 332 genes"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a>score.df <span class="ot">&lt;-</span> <span class="fu">rbind</span>(score.df,score.ISG)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>In the next table we can observe how many marker genes we defined and how may are detected in the sample.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a>score.df[,<span class="fu">c</span>(<span class="st">"N. detected"</span>,<span class="st">"N. total"</span>)]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                N. detected N. total
Epithelial genes                          2       42
Enterocytes                               0        4
Goblet cells                              0        6
Enteroendocrine cells                     0        4
Paneth cells                              1        3
Tuft cells                                0        3
TA cells                                  2        5
Stem cells                                1        5
BEST4 cells                               1        4
Red blood genes                           3        3
T cell genes                              6        6
B cell genes                              6        6
Plasma Cells                              2        4
Dendridic Cells                           7        8
Macrophages                               5        5
ILCs - Cytotoxic natural killer           8        8
ILC1s                                    10       11
ILC2s                                     5       11
ILC3                                      1        6
Mast cells                                1        6
Endotelial Cells                          2        7
??                                        0        2
Fibroblasts                               1        6
Smooth muscle cells                       3        5
Enteric glia cells                        2        4
Neutrophils                               5        8
Eosinophil                                0        2
Basophils                                 2        6
ISG                                     184      332</code></pre>
</div>
</div>
<p>Cluster tree using the cosine distance over cluster coex.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a>cluster_data <span class="ot">&lt;-</span> <span class="fu">getClusterizationData</span>(obj)[[<span class="st">"coex"</span>]]</span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-3"><a href="#cb60-3" aria-hidden="true" tabindex="-1"></a><span class="do">######## This is the best: cosine dissimilarity</span></span>
<span id="cb60-4"><a href="#cb60-4" aria-hidden="true" tabindex="-1"></a>Matrix <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">t</span>(cluster_data))</span>
<span id="cb60-5"><a href="#cb60-5" aria-hidden="true" tabindex="-1"></a>sim <span class="ot">&lt;-</span> Matrix <span class="sc">/</span> <span class="fu">sqrt</span>(<span class="fu">rowSums</span>(Matrix <span class="sc">*</span> Matrix))</span>
<span id="cb60-6"><a href="#cb60-6" aria-hidden="true" tabindex="-1"></a>sim <span class="ot">&lt;-</span> sim <span class="sc">%*%</span> <span class="fu">t</span>(sim)</span>
<span id="cb60-7"><a href="#cb60-7" aria-hidden="true" tabindex="-1"></a>D_sim <span class="ot">&lt;-</span> <span class="fu">as.dist</span>(<span class="dv">1</span> <span class="sc">-</span> sim)</span>
<span id="cb60-8"><a href="#cb60-8" aria-hidden="true" tabindex="-1"></a>tree <span class="ot">&lt;-</span> <span class="fu">hclust</span>(D_sim,<span class="at">method =</span> <span class="st">"ward.D2"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get_col(col, k): Length of color vector was longer than the number of
clusters - first k elements are used</code></pre>
</div>
</div>
<div class="cell">

</div>
<div class="cell">

</div>
</section>
<section id="final-score-heatmap-2" class="level3">
<h3 class="anchored" data-anchor-id="final-score-heatmap-2">Final score heatmap</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a>col_fun <span class="ot">=</span> <span class="fu">colorRamp2</span>(<span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>), <span class="fu">c</span>( <span class="st">"lightblue"</span>, <span class="st">"red"</span>))</span>
<span id="cb62-2"><a href="#cb62-2" aria-hidden="true" tabindex="-1"></a>final.heatmap <span class="ot">&lt;-</span> <span class="fu">Heatmap</span>(df.t, <span class="at">rect_gp =</span> <span class="fu">gpar</span>(<span class="at">col =</span> <span class="st">"white"</span>, <span class="at">lwd =</span> <span class="dv">1</span>),</span>
<span id="cb62-3"><a href="#cb62-3" aria-hidden="true" tabindex="-1"></a>        <span class="co">#cluster_rows = F,</span></span>
<span id="cb62-4"><a href="#cb62-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">cluster_columns =</span> <span class="cn">FALSE</span>,</span>
<span id="cb62-5"><a href="#cb62-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">show_heatmap_legend =</span> F,</span>
<span id="cb62-6"><a href="#cb62-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">col =</span> col_fun,</span>
<span id="cb62-7"><a href="#cb62-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">width =</span>  <span class="fu">unit</span>(<span class="dv">28</span>, <span class="st">"cm"</span>),</span>
<span id="cb62-8"><a href="#cb62-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">row_dend_width =</span> <span class="fu">unit</span>(<span class="dv">8</span>, <span class="st">"cm"</span>),</span>
<span id="cb62-9"><a href="#cb62-9" aria-hidden="true" tabindex="-1"></a>        <span class="co">#height = unit(6, "cm"),</span></span>
<span id="cb62-10"><a href="#cb62-10" aria-hidden="true" tabindex="-1"></a>        <span class="at">cluster_rows =</span> dend,</span>
<span id="cb62-11"><a href="#cb62-11" aria-hidden="true" tabindex="-1"></a>        <span class="at">column_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">11</span>),</span>
<span id="cb62-12"><a href="#cb62-12" aria-hidden="true" tabindex="-1"></a>      <span class="at">row_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">9</span>),</span>
<span id="cb62-13"><a href="#cb62-13" aria-hidden="true" tabindex="-1"></a>        <span class="at">cell_fun =</span> <span class="cf">function</span>(j, i, x, y, width, height, fill) {</span>
<span id="cb62-14"><a href="#cb62-14" aria-hidden="true" tabindex="-1"></a>        <span class="fu">grid.text</span>(<span class="fu">sprintf</span>(<span class="st">"%.1f"</span>, df.t[i, j]), x, y, <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">9</span>))</span>
<span id="cb62-15"><a href="#cb62-15" aria-hidden="true" tabindex="-1"></a>},</span>
<span id="cb62-16"><a href="#cb62-16" aria-hidden="true" tabindex="-1"></a><span class="at">right_annotation =</span> ha,</span>
<span id="cb62-17"><a href="#cb62-17" aria-hidden="true" tabindex="-1"></a><span class="at">left_annotation =</span> hb</span>
<span id="cb62-18"><a href="#cb62-18" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb62-19"><a href="#cb62-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-20"><a href="#cb62-20" aria-hidden="true" tabindex="-1"></a>final.heatmap</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="04_Data_exploration_clusters_id_patient5_NB_files/figure-html/unnamed-chunk-41-1.png" class="img-fluid" width="1536"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb63"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(final.heatmap,<span class="fu">paste0</span>(<span class="st">"../../Crohn_disease/data/04_cluster_identification/"</span>,sample.id,<span class="st">"_heatmap.RDS"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="session-info" class="level6">
<h6 class="anchored" data-anchor-id="session-info">Session info</h6>
<div class="cell">
<div class="sourceCode cell-code" id="cb64"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>R version 4.2.2 Patched (2022-11-10 r83330)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.5 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0

locale:
 [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       
 [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   
 [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          
[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
[1] COTAN_2.0.0       testthat_3.1.6    dendextend_1.16.0 stringr_1.4.0    
[5] ggplot2_3.4.0     viridis_0.6.2     viridisLite_0.4.0 gridExtra_2.3    

loaded via a namespace (and not attached):
  [1] readxl_1.4.0          circlize_0.4.15       plyr_1.8.7           
  [4] igraph_1.3.4          lazyeval_0.2.2        sp_1.5-0             
  [7] splines_4.2.2         listenv_0.8.0         scattermore_0.8      
 [10] usethis_2.1.6         digest_0.6.29         foreach_1.5.2        
 [13] htmltools_0.5.2       fansi_1.0.3           magrittr_2.0.3       
 [16] memoise_2.0.1         tensor_1.5            cluster_2.1.4        
 [19] doParallel_1.0.17     ROCR_1.0-11           remotes_2.4.2        
 [22] ComplexHeatmap_2.13.0 globals_0.16.2        matrixStats_0.63.0   
 [25] spatstat.sparse_3.0-0 prettyunits_1.1.1     colorspace_2.0-3     
 [28] ggrepel_0.9.2         xfun_0.31             dplyr_1.0.9          
 [31] callr_3.7.0           crayon_1.5.1          jsonlite_1.8.0       
 [34] spatstat.data_3.0-0   progressr_0.11.0      survival_3.4-0       
 [37] zoo_1.8-10            iterators_1.0.14      glue_1.6.2           
 [40] polyclip_1.10-0       gtable_0.3.0          leiden_0.4.2         
 [43] GetoptLong_1.0.5      pkgbuild_1.4.0        RcppZiggurat_0.1.6   
 [46] future.apply_1.10.0   shape_1.4.6           BiocGenerics_0.43.0  
 [49] abind_1.4-5           scales_1.2.0          DBI_1.1.3            
 [52] ggthemes_4.2.4        spatstat.random_3.0-1 miniUI_0.1.1.1       
 [55] Rcpp_1.0.9            xtable_1.8-4          clue_0.3-61          
 [58] spatstat.core_2.4-4   reticulate_1.25       stats4_4.2.2         
 [61] htmlwidgets_1.5.4     httr_1.4.4            RColorBrewer_1.1-3   
 [64] ellipsis_0.3.2        Seurat_4.2.0          factoextra_1.0.7     
 [67] ica_1.0-2             farver_2.1.0          pkgconfig_2.0.3      
 [70] uwot_0.1.14           deldir_1.0-6          utf8_1.2.2           
 [73] labeling_0.4.2        tidyselect_1.2.0      rlang_1.0.6          
 [76] reshape2_1.4.4        later_1.3.0           cellranger_1.1.0     
 [79] munsell_0.5.0         tools_4.2.2           cachem_1.0.6         
 [82] cli_3.4.1             gsubfn_0.7            generics_0.1.2       
 [85] devtools_2.4.3        ggridges_0.5.4        evaluate_0.18        
 [88] fastmap_1.1.0         goftest_1.2-3         yaml_2.3.5           
 [91] processx_3.7.0        knitr_1.40            fs_1.5.2             
 [94] fitdistrplus_1.1-8    purrr_0.3.4           RANN_2.6.1           
 [97] nlme_3.1-160          pbapply_1.6-0         future_1.29.0        
[100] mime_0.12             brio_1.1.3            compiler_4.2.2       
[103] rstudioapi_0.13       plotly_4.10.0         png_0.1-7            
[106] spatstat.utils_3.0-1  tibble_3.1.7          stringi_1.7.6        
[109] ps_1.7.0              desc_1.4.1            rgeos_0.5-9          
[112] lattice_0.20-45       Matrix_1.5-1          vctrs_0.5.0          
[115] pillar_1.8.1          lifecycle_1.0.3       spatstat.geom_3.0-3  
[118] lmtest_0.9-40         GlobalOptions_0.1.2   RcppAnnoy_0.0.19     
[121] data.table_1.14.2     cowplot_1.1.1         irlba_2.3.5.1        
[124] httpuv_1.6.5          patchwork_1.1.1       R6_2.5.1             
[127] promises_1.2.0.1      KernSmooth_2.23-20    IRanges_2.31.2       
[130] parallelly_1.32.1     sessioninfo_1.2.2     codetools_0.2-18     
[133] MASS_7.3-58           assertthat_0.2.1      pkgload_1.3.0        
[136] proto_1.0.0           rprojroot_2.0.3       rjson_0.2.21         
[139] withr_2.5.0           SeuratObject_4.1.2    sctransform_0.3.5    
[142] S4Vectors_0.35.0      mgcv_1.8-41           parallel_4.2.2       
[145] rpart_4.1.19          tidyr_1.2.0           rmarkdown_2.17       
[148] Rfast_2.0.6           Rtsne_0.16            shiny_1.7.1          </code></pre>
</div>
</div>


</section>
</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      let href = ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
</div> <!-- /content -->



</body></html>